services/graph/pkg/identity/ldap.go

```diff
@@ -254,14 +298,34 @@ func (i *LDAP) UpdateUser(ctx context.Context, nameOrID string, user libregraph.
 		}
 	}
 
+	// Behavior of enabling/disabling of users depends on the "disableUserMechanism" config option:
+	//
+	// "attribute": For the upstream user management service which modifies accountEnabled on the user entry
+	// "group": Makes it possible for local admins to disable users by adding them to a special group
 	if user.AccountEnabled != nil {
-		boolString := strings.ToUpper(strconv.FormatBool(*user.AccountEnabled))
-		ldapValue := e.GetEqualFoldAttributeValue(i.userAttributeMap.accountEnabled)
-		updateNeeded = true
-		if ldapValue != "" {
-			mr.Replace(i.userAttributeMap.accountEnabled, []string{boolString})
-		} else {
-			mr.Add(i.userAttributeMap.accountEnabled, []string{boolString})
+		switch i.disableUserMechanism {
+		case DisableMechanismNone:
+			return nil, errors.New("accountEnabled option not compatible with backend disable user mechanism")
```

Comment:
Same question as above, do we really want to error based on the local configuration, making using the API more complex?